### PR TITLE
Allow initdb to be re-run

### LINKED ISSUE #4220: Running initdb twice causes a traceback
If you somehow manage to run ``initdb`` twice you will get the following error.  I'm not sure if it is possible to detect that initdb has already been run and bail out.

```pytb
(python2.7) $> pootle initdb
2015-11-25 01:02:41,471 INFO Loading custom settings from '/home/arashm/.pootle/pootle.conf'...
2015-11-24 21:32:41,699 INFO Using Python PO
Populating the database.
Traceback (most recent call last):
  File "/home/arashm/.pyenvs/python2.7/bin/pootle", line 9, in <module>
    load_entry_point('Pootle==2.7.3.dev20151124100802', 'console_scripts', 'pootle')()
  File "/home/arashm/workspace/Mozilla/pootle/pootle/runner.py", line 319, in main
    django_settings_module='pootle.settings')
  File "/home/arashm/workspace/Mozilla/pootle/pootle/runner.py", line 299, in run_app
    management.execute_from_command_line(command)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/core/management/__init__.py", line 385, in execute_from_command_line
    utility.execute()
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/core/management/__init__.py", line 377, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/core/management/base.py", line 288, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/core/management/base.py", line 338, in execute
    output = self.handle(*args, **options)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/core/management/base.py", line 533, in handle
    return self.handle_noargs(**options)
  File "/home/arashm/workspace/Mozilla/pootle/pootle/apps/pootle_app/management/commands/initdb.py", line 25, in handle_noargs
    initdb()
  File "/home/arashm/workspace/Mozilla/pootle/pootle/core/initdb.py", line 35, in initdb
    create_default_projects()
  File "/home/arashm/workspace/Mozilla/pootle/pootle/core/initdb.py", line 255, in create_default_projects
    tutorial.save()
  File "/home/arashm/workspace/Mozilla/pootle/pootle/apps/pootle_project/models.py", line 368, in save
    super(Project, self).save(*args, **kwargs)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/models/base.py", line 589, in save
    force_update=force_update, update_fields=update_fields)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/models/base.py", line 617, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/models/base.py", line 698, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/models/base.py", line 731, in _do_insert
    using=using, raw=raw)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/models/manager.py", line 92, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/models/query.py", line 921, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 921, in execute_sql
    cursor.execute(sql, params)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/backends/utils.py", line 81, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/home/arashm/.pyenvs/python2.7/lib/python2.7/site-packages/django/db/backends/sqlite3/base.py", line 485, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.IntegrityError: UNIQUE constraint failed: pootle_app_project.directory_id
```

### PR DESCRIPTION
**Pull request by phlax**
Also adds debug logging to initdb

Fixes #4220

### CODE REVIEW

File: pootle/core/initdb.py [hunk 1]
```diff
@@ -213,8 +223,7 @@ def create_template_languages():
     The 'templates' language is used to give users access to the untranslated
     template files.
     """
-    templates, created = Language.objects.get_or_create(code="templates",
-                                                        fullname=u'Templates')
+    _create_object(Language, **dict(code="templates", fullname="Templates"))
```
Comment (unho) [inline pootle/core/initdb.py]: Having a reference like `**criteria` looks fine to me, but creating the whole dict here is less readable. I suggest:

```python
    criteria = {
        'code': 'templates',
        'fullname': 'Templates',
    }
    _create_object(Language, **criteria)
```

The same for the other `**dict` appearances.

File: pootle/core/initdb.py [hunk 2]
```diff
@@ -45,7 +78,6 @@ def create_essential_users():
 
     These users are required for Pootle's permission system.
     """
-    User = get_user_model()
```
Comment (unho) [inline pootle/core/initdb.py]: Nit: You can also remove the blank line after this.

File: pootle/core/initdb.py (hunk 1 above)
Comment (phlax) [inline pootle/core/initdb.py]: i dont agree - if anything i would change the other to more like:

```
_create_object(
    Language,
    **{'code': 'templates',
        'fullname': 'Templates'})
```

File: pootle/core/initdb.py [hunk 3]
```diff
@@ -36,6 +41,34 @@ def initdb():
...
+def _create_pootle_user(**criteria):
+    user = _create_object(get_user_model(), **criteria)
+    user.set_unusable_password()
+    user.save()
+    return user
+
+
+def _create_pootle_permission_set(permissions, **criteria):
+    permission_set = _create_object(PermissionSet, **criteria)
+    if permission_set.positive_permissions != permissions:
```
Comment (unho) [inline pootle/core/initdb.py]: What is the reason of this check? Is it not ok to just assign the `positive_permissions`?

Comment (phlax) [inline pootle/core/initdb.py]: yep - just for efficiency really - to avoid resaving the instance unnecessarily

Comment (unho) [inline pootle/core/initdb.py]: Actually just noticed that we want to check if all the specified `permissions` are on `permission_set.positive_permissions` and add the missing ones (while keeping any other extra permission there), while here we are checking that `permission_set.positive_permissions` is exactly `permissions`. Also blindly setting it to `permissions` is not an option.

Comment (phlax) [inline pootle/core/initdb.py]: k - good spot

File: pootle/core/initdb.py (hunk 1 above)
Comment (unho) [inline pootle/core/initdb.py]: Then just keep at it is now.

File: pootle/core/initdb.py [hunk 4]
```diff
@@ -36,6 +41,34 @@ def initdb():
...
 
+def _create_object(model_klass, **criteria):
+    instance, created = model_klass.objects.get_or_create(**criteria)
+    if created:
+        logger.debug(
+            "Created %s: '%s'"
+            % (instance.__class__.__name__, instance))
+    else:
+        logger.debug(
+            "%s exists - skipping: '%s'"
```
Comment (unho) [inline pootle/core/initdb.py]: Nit: %s **already** exists - skipping ...

File: pootle/core/initdb.py (hunk 2 above)
Comment (phlax) [inline pootle/core/initdb.py]: removed

**Comment by unho** [comment]
lgtm

**Pull request merged**